### PR TITLE
Trigger subsonicupdate on database and smart playlist changes

### PR DESCRIPTION
**Pull request by JOJ0**
## Description

- Fixes triggering subsonicupdate on a database_change event instead of only at an import event.
- Sends a new event from the smartplaylist plugin whenever lists are updated/created and has subsonicupdate plugin listen to it.
- Both events register new listeners that launch the actual Subsonic library update _at the very end of a beets run_ (cli_exit) instead of being triggered by each change (similar to how the mpdupdate plugin does it).
- It doesn't hurt if Subsonic is informed on subsequent changes of smart playlists (while the user is heavily experimenting with them) since once a subsonic-rescan is triggered, it keeps running and doesn't restart from scratch.

## To Do

- [x] Documentation.
- [x] Changelog.
- [x] ~Tests.~

### CODE REVIEW

**Review by sampsyo — APPROVED** [review]
Interesting; this seems pretty good! One odd thing to note is that I don't think we have ever emitted an even _from a plugin_ before. (Grepping for `send_event` in the beetsplug directory yields only false-positives in the `bpd` plugin.) I think it's fine for this to happen, but I just thought it was worth remarking on the fact that this is the first instance of plugin-to-plugin communication.

**Comment by JOJ0** [comment]
Oh wow! That is indeed interesting. I wouldn't have thought of that this was never used before. As usual, thanks a lot for the insight!

One thing I was wondering while putting this PR together was: "Should I include the same feature for mpd/bpd plugins?" I didn't in the end because I never felt that for my use-cases mpd requires to be informed about playlist changes. But I'm not a heavy mpd user, and have a rather unusual setup.

tl;dr; I would be interested what you think about it. Would it make sense to receive this event in the `mpdupdate` plugin?

**Comment by sampsyo** [comment]
I guess I think it's OK to just add this one feature! No particular need to preemptively cover other plugins… if a user of bpd requests this behavior there, we can consider it then.
